cmd/docker-mcp/internal/gateway/run.go

```diff
@@ -200,19 +164,20 @@ func (g *Gateway) listServersAndCapabilities(ctx context.Context, configuration
 	log("- Listing MCP tools...")
 	capabilities, err := g.listCapabilities(ctx, configuration, serverNames)
 	if err != nil {
-		return nil, fmt.Errorf("listing resources: %w", err)
+		return fmt.Errorf("listing resources: %w", err)
 	}
 	log(">", len(capabilities.Tools), "tools listed in", time.Since(startList))
 
-	return capabilities, nil
-}
-
-func refreshCapabilities(s *server.MCPServer, c *Capabilities) {
-	s.SetTools(c.Tools...)
-	s.SetPrompts(c.Prompts...)
-	s.SetResources(c.Resources...)
-	s.RemoveAllResourceTemplates()
-	for _, v := range c.ResourceTemplates {
-		s.AddResourceTemplate(v.ResourceTemplate, v.Handler)
+	// Update the server's capabilities.
+	g.health.SetUnhealthy()
+	mcpServer.SetTools(capabilities.Tools...)
```

Comment:
I'm actually kind of worried about the conflicts here because we could end up with conflicting tools appearing and disappearing depending on which server last changed.